Fixed a crash for podcastservice

Rovanion opened the pull request:
If the filespaths stored in the podcastDoa DB pointed to paths which the server could not open, the whole server crashed. Fixed this little error by, somewhat naively, catching all the exceptions thrown and removing the offending episodes from the DB.

Rovanion committed:
Added a probably overly simple exception handling for loading podcasts

Commit:
Fixed fail to start if podcast episodes weren't where the db expected them to be

Commit:
Merge branch 'master' of github.com:Rovanion/supersonic

Conflicts:
	subsonic-main/src/main/java/net/sourceforge/subsonic/service/PodcastService.java

timoreimann commented:
I believe your fix should be improved in the following regards:
1. Catching the base Exception is too generic and not necessary: [MediaFileService's getMediaFile()](https://github.com/Rovanion/supersonic/blob/master/subsonic-main/src/main/java/net/sourceforge/subsonic/service/MediaFileService.java#L86) mentions in its Javadoc portion that it's throwing a **SecurityException** _if access is denied to the given file_. So catching that more specific exception should suffice.
2. If I understand the code path correctly, the error will only occur after the podcast file had been stored on disk successfully before. This means that whatever happened that now prevents the server from accessing the file wasn't a problem in the first place, likely indicating that this is a temporary error. Concludingly, I don't think that removing the episode from the database permanently is the best approach. Instead, it should be kept and the user be notified of the issue (or, if that is too much effort for the moment, at least evict a log message of level _WARNING_).

Would you agree to that improvement? If so, could you update your feature branch accordingly?

Rovanion commented:
1. Yes, I made the catch overly naïve because I didn't know what exceptions other than SecurityException could be thrown at it. Now if that's the only thing which can be thrown from getMediaFile() then it's of course better to catch that specifically.
2. In my case I moved the podcast folder from one path to another and changed the setting in Supersonic accordingly. Now apparently this didn't change the paths in the DB making the server crash on start. Atleast in my case, removing the episodes from the DB permanently and letting the scan detect them again in the new folder made perfect sense.

I thought i had a log message going out when I removed the files. Seems it gost lost somewhere. Will fix that and the exception.

Commit:
Missed log message and narrowed down Exception handling.

Commit:
Changed log type to warn.

timoreimann commented:
So I understand that you caused the access problems yourself in the first place and had Supersonic fix it by deleting the old DB entry and letting the new directory / file location get rescanned. I'm almost fine with that approach except for wondering whether you'll lose any information stored about the podcast in the database that you had before, such as usage statistics. I must say that I haven't used podcasts at all in Supersonic, so I appreciate any feedback you can provide on this.

Still, getting a crash fix first is the major goal, and the approach you've taken seems to do it. 

--Timo

Rovanion commented:
Well in all fairness this is only fixing the symptoms, not the actual issue. The actual issue would be that when the path to the podcast folder is changed in the settings, the path in the DB is not.

So the best course of action, I think, would be to keep my current changes to the code in. But more than that make sure that the paths in the DB are actually changed when the setting is changed. This way my remove-from-DB code will hopefully never be triggered by my specific course of action.

But the remove-from-DB code will still be triggered, and prevent a crash, in another course of actions where the user for example removes a single episode from the podcast folder.

timoreimann commented:
Agreed. I'll open up another issue to make sure we won't forget about the "real" fix.
